sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -135,7 +155,19 @@ public Resource merge(@Nullable Resource other) {
     AttributesBuilder attrBuilder = Attributes.builder();
     attrBuilder.putAll(this.getAttributes());
     attrBuilder.putAll(other.getAttributes());
-    return new AutoValue_Resource(attrBuilder.build());
+
+    if (other.getSchemaUrl() == null) {
+      return create(attrBuilder.build(), getSchemaUrl());
+    }
+    if (getSchemaUrl() == null) {
+      return create(attrBuilder.build(), other.getSchemaUrl());
+    }
+    if (!Objects.equals(other.getSchemaUrl(), getSchemaUrl())) {
```

Comment:
Perhaps we should log a warning here? Users can be surprised by missing schema knowing it was set before the merge.

Reply:
We can use normal `String.equals` since we checked both for nulls

Reply:
I think it's reasonable to log here, until we have a strategy for merging them, yes. I'll add it.

Reply:
done